.github/workflows/olm.yaml

```diff
@@ -0,0 +1,87 @@
+---
+name: OLM
+
+on:
+  push:
+    branches: [auto-release, main]
+    tags: ["v*"]
+  pull_request:
+    branches: [auto-release]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    environment: quay
+
+    steps:
+      # setup
+      - uses: actions/checkout@v2
+
+      - name: Initialize ENV vars
+        run: |
+          cat $GITHUB_ENV
+
+          echo "GITHUB_REF: $GITHUB_REF"
+
+          SOURCE_NAME=${GITHUB_REF#refs/*/}
+          SOURCE_BRANCH=${GITHUB_REF#refs/heads/}
+          SOURCE_TAG=${GITHUB_REF#refs/tags/}
+
+          if [[ $SOURCE_NAME == auto-release ]]; then
+            VERSION="0.0.1-$(date +%y%m%d%H%M%S)"
+            CHANNELS=candidate
+          else
+            VERSION=${SOURCE_NAME##v}
+            CHANNELS=candidate,stable
+          fi
+
+          IMG_BASE=$(grep '^IMAGE_TAG_BASE' -- Makefile | cut -f3 -d ' ')
+
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+          echo "CHANNELS=$CHANNELS" >> $GITHUB_ENV
+          echo "IMG=${IMG_BASE}:$VERSION" >> $GITHUB_ENV
+          echo "BUNDLE_IMG=${IMG_BASE}-bundle:$VERSION" >> $GITHUB_ENV
+          echo "CATALOG_IMG=${IMG_BASE}-catalog:latest" >> $GITHUB_ENV
+          echo ---------------------------
+          cat $GITHUB_ENV
+          echo ---------------------------
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v2.1.4
+        with:
+          go-version: 1.16
+
+      - name: Build Operator
+        run: make operator
+
+      - name: Registry Login
+        # You may pin to the exact commit or the version. uses: docker/login-action@v1.10.0
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_LOGIN }}
+          password: ${{ secrets.QUAY_TOKEN }}
+
+      - name: Build Operator Image
+        run: make image
+
+      - name: Publish Operator
+        run: make image-push
+
+      - name: Bundle
+        run: make bundle
+
+      - name: Build Bundle Image
+        run: make bundle-image
+
+      - name: Publish Bundle
+        run: make bundle-push
+
+
+      - name: Build Catalog Image
+        run: |
+          env | grep CAT
```

Comment:
Debug statement?

Reply:
indeed, and I would like to leave that on until we have a stable release. Not knowing what these vars are did make the debugging of an CI failure hard